src/v/cluster/producer_state.h

```diff
@@ -209,6 +209,17 @@ class producer_state {
     void update_current_txn_start_offset(std::optional<kafka::offset> offset) {
         _current_txn_start_offset = offset;
     }
+
+    model::producer_identity id() const { return _id; }
```

Comment:
Nit: I guess this better fits in https://github.com/redpanda-data/redpanda/pull/22781/commits/a5bb2d2ea4f19eb5fa047254e2fc613727683869?